src/Microsoft.Sbom.Parsers.Spdx22SbomParser/Parser/SbomFileParser.cs

```diff
@@ -0,0 +1,294 @@
+        }
+
+        if (string.IsNullOrEmpty(sbomFile.SPDXId))
+        {
+            missingProps.Add(nameof(sbomFile.SPDXId));
+        }
+
+        if (string.IsNullOrEmpty(sbomFile.FileCopyrightText))
+        {
+            missingProps.Add(nameof(sbomFile.FileCopyrightText));
+        }
+
+        if (string.IsNullOrEmpty(sbomFile.LicenseConcluded))
+        {
+            missingProps.Add(nameof(sbomFile.LicenseConcluded));
+        }
+
+        if (sbomFile.LicenseInfoInFiles == null || sbomFile.LicenseInfoInFiles.Count == 0)
+        {
+            missingProps.Add(nameof(sbomFile.LicenseInfoInFiles));
+        }
+
+        if (missingProps.Count() > 0)
+        {
+            throw new ParserError($"Missing required value(s) for file object at position {stream.Position}: {string.Join(",", missingProps)}");
+        }
+    }
+
+    private void ParseProperty(ref Utf8JsonReader reader, ref byte[] buffer)
+    {
+        switch (reader.GetString())
+        {
+            case "fileName":
+                ParserUtils.Read(stream, ref buffer, ref reader);
+                sbomFile.Path = ParseNextString(ref reader, ref buffer);
+                break;
+
+            case "SPDXID":
+                ParserUtils.Read(stream, ref buffer, ref reader);
+                sbomFile.SPDXId = ParseNextString(ref reader, ref buffer);
+                break;
+
+            case "checksums":
+                ParserUtils.Read(stream, ref buffer, ref reader);
+                sbomFile.Checksum = ParseChecksumsArray(ref reader, ref buffer);
+                break;
+
+            case "licenseConcluded":
+                ParserUtils.Read(stream, ref buffer, ref reader);
+                sbomFile.LicenseConcluded = ParseNextString(ref reader, ref buffer);
+                break;
+
+            case "copyrightText": 
+                ParserUtils.Read(stream, ref buffer, ref reader);
+                sbomFile.FileCopyrightText = ParseNextString(ref reader, ref buffer);
+                break;
+
+            case "licenseInfoInFiles":
+                ParserUtils.Read(stream, ref buffer, ref reader);
+                sbomFile.LicenseInfoInFiles = ParseLicenseInfoInFilesArray(ref reader, ref buffer);
+                break;
+
+            default:
+                SkipProperty(ref reader, ref buffer);
+                break;
+        }
+    }
+
+    private List<string> ParseLicenseInfoInFilesArray(ref Utf8JsonReader reader, ref byte[] buffer)
+    {
+        var licenses = new List<string>();
+
+        // Read the opening [ of the array
+        ParserUtils.AssertTokenType(stream, ref reader, JsonTokenType.StartArray);
+
+        while (reader.TokenType != JsonTokenType.EndArray)
+        {
+            ParserUtils.Read(stream, ref buffer, ref reader);
+            if (reader.TokenType == JsonTokenType.EndArray)
+            {
+                break;
+            }
+
+            licenses.Add(reader.GetString());
```

Comment:
You could also have the following code to avoid the break:
if (reader.TokenType == JsonTokenType.EndArray)
{
       licenses.Add(reader.GetString());
}

Reply:
The licenses.Add method and the line before this == both read to the buffer which is a mutable call and I added the break just to avoid this.